Tests/Marketplace/marketplace_services.py

```diff
@@ -565,6 +568,92 @@ def upload_to_storage(self, zip_pack_path, latest_version, storage_bucket, overr
+        :return: Tuple of integers.
+        """
+        if path:
+            major, minor, rev = re.search('(\d+)_(\d+)_(\d+)', version).groups()
+        else:
+            major, minor, rev = re.search('(\d+)\.(\d+)\.(\d+)', version).groups()
+        return int(major), int(minor), int(rev)
+
+    def prepare_release_notes(self, storage_bucket):
+        """Need to implement the changelog.md parsing and changelog.json creation after design is
+        finalized.
+
+        """
+        task_status = False
+        try:
+            metadata_path = os.path.join(self._pack_path, Pack.METADATA)
+            with open(metadata_path, "r") as metadata_file:
+                metadata_json = json.load(metadata_file)
+            expected_release_version_str = metadata_json.get('currentVersion')
+            expected_release_version_int = self.major_minor_rev(expected_release_version_str, path=False)
+            release_notes_dir = os.path.join(self._pack_path, Pack.RELEASE_NOTES)
+            found_versions = []
+            for filename in os.listdir(release_notes_dir):
+                found_versions.append(filename)
+            print_warning(f"Found versions are: {found_versions}")
+            if len(found_versions) == 0:
+                raise Exception(f"Release notes not found for {self._pack_name}. Please add them under the "
+                                f"ReleaseNotes directory.")
+            elif len(found_versions) == 1:
+                release_notes = {}
+                # If the found versions are only 1, there won't be a releaseNotes.json file in GCP
+                version_changelog = {'releaseNotes': 'HERE ARE RELEASE NOTES FOR FIRST RELEASE',
+                                     'released': datetime.utcnow().strftime(Metadata.DATE_FORMAT)}
+                release_notes['1.1.1'] = version_changelog
+                # with open(os.path.join(self._pack_path, 'releaseNotes.json'), "w") as f:
+                #     json.dump(f, release_notes)
+            else:
+            # if os.path.exists(os.path.join(self._pack_path, 'changelog.json')):
+                print_warning("Path exists")
+                latest = max(found_versions, key=self.major_minor_rev)
+                latest_int = self.major_minor_rev(latest)
+                _version = str(latest).replace('.md', '')
+                version = str(_version).replace('_', '.')
+                if not expected_release_version_int == latest_int:
+                    raise Exception(f"Conflict between version found in metadata ({expected_release_version_str}) "
+                                    f"and latest version found in ReleaseNotes dir ({version})")
+                    # sys.exit(1)
+
+                # We need to retrieve previous releaseNotes.json here
+                latest_gcp_rn_path = os.path.join(GCPConfig.STORAGE_BASE_PATH, self._pack_name, _version, '.json')
+                existing_rn_file = storage_bucket.blob(prefix=latest_gcp_rn_path)
```

Comment:
Wrong logic. You have all versions of release notes in `index` (`changelog.json` file).
Similar to what @liorblob did with metadata versions.